dask_kubernetes/common/networking.py

```diff
@@ -76,12 +89,14 @@ def _random_free_port(low, high, retries=20):
 
 async def port_forward_service(service_name, namespace, remote_port, local_port=None):
     check_dependency("kubectl")
-    if not local_port:
+    if not local_port or _port_in_use(local_port):
```

Comment:
If the port is already in use we should raise a warning, otherwise the user may be confused why they've specified a port but it isn't using it.